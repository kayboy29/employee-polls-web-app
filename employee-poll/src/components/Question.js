import React from "react";
import { Link } from "react-router-dom";

const Question = (props) => {
  return (
    <div className="question__card">
      <p className="question__author" data-testid="question-author">
        {props.question.author}
      </p>
      <p className="question__date" data-testid="question-date">
        {props.question.timestamp}
      </p>
      <p>
        <Link to={`/questions/${props.question.id}`}>Show</Link>
      </p>
    </div>
  );
};

export default Question;
