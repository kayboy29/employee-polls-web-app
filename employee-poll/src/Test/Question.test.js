import Question from "../components/Question";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import React from "react";

describe("Question", () => {
  it("matches the snapshot when question is passed", () => {
    var question = {
      id: "xj352vofupe1dqz9emx13r",
      author: "mtsamis",
      timestamp: 1493579767190,
    };
    var component = render(
      <MemoryRouter>
        <Question question={question} />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("will display all fields when question is passed", () => {
    var question = {
      id: "xj352vofupe1dqz9emx13r",
      author: "mtsamis",
      timestamp: 1493579767190,
    };
    var component = render(
      <MemoryRouter>
        <Question question={question} />
      </MemoryRouter>
    );
    expect(component.getByTestId("question-author")).toBeInTheDocument();
    expect(component.getByTestId("question-date")).toBeInTheDocument();
  });
});
