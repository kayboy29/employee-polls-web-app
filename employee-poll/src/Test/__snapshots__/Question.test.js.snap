// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question matches the snapshot when question is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="question__card"
      >
        <p
          class="question__author"
          data-testid="question-author"
        >
          mtsamis
        </p>
        <p
          class="question__date"
          data-testid="question-date"
        >
          1493579767190
        </p>
        <p>
          <a
            href="/questions/xj352vofupe1dqz9emx13r"
          >
            Show
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="question__card"
    >
      <p
        class="question__author"
        data-testid="question-author"
      >
        mtsamis
      </p>
      <p
        class="question__date"
        data-testid="question-date"
      >
        1493579767190
      </p>
      <p>
        <a
          href="/questions/xj352vofupe1dqz9emx13r"
        >
          Show
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
